File: CookingWebsite/CookingWebsite.Application/Dtos/RecipeDtos/AddRecipeDto.cs

namespace CookingWebsite.Application.Dtos.RecipeDtos
{
    public class AddRecipeDto
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public int CookingTime { get; set; }
        public int PersonsCount { get; set; }
        public string AuthorUsername { get; set; }
        public RecipeIngredientDto recipeIngredient { get; set; }
    }
}

File: CookingWebsite/CookingWebsite/ClientApp/src/app/pages/recipe-create/add-recipe-dto.ts
import { RecipeIngredientDto } from "../../Dtos/recipe-ingredient-dto";
import { AddRecipeStepDto } from "./add-recipe-step-dto";

export interface AddRecipeDto {
  image: File;

  imageUrl: string;

  authorUsername: string

  title: string;
  description: string;

  tags: string[];
  tagsString: string;

  cookingTime: number;
  personsCount: number;

  ingredients: RecipeIngredientDto[];
  steps: AddRecipeStepDto[];
}
